thriftbp/client_pool.go

```diff
@@ -303,7 +255,17 @@ type clientPool struct {
 	releaseErrorCounter  metrics.Counter
 }
 
-func (p *clientPool) GetClient() (Client, error) {
+func (p *clientPool) Call(ctx context.Context, method string, args, result thrift.TStruct) error {
+	client, err := p.getClient()
+	if err != nil {
+		return fmt.Errorf("thriftbp: error retrieving thrift client from pool. %w", err)
```

Comment:
I think any error returned by `Call` will be wrapped by thrift library code again, so here we might want:

1. Define this as a type to be inspectable (but I also need to make an upstream change to make sure the thrift wrapped errors has `Unwrap` implemented, currently they don't)
2. Since we also do error logging in `releaseClient`, we probably should do error logging here as well.

Reply:
`%w` makes it unwrappable :)

Reply:
yes but it won't work if it's wrapped again (by `TApplicationException` or `TTransportException`) and that wrapping is not unwrappable.

Reply:
Fair, it's also _probably_ fine to just not do this.

Reply:
Also what I meant is we want to make it easy for the service code to distinguish between errors caused by unable to get a client from the pool, and errors caused by the actual call itself.

Reply:
Just checked thrift code and error returned by `Call` is returned to the caller directly without further wrapping, so defining `PoolError` is sufficient.